pkg/ottl/ottlfuncs/func_index.go

```diff
@@ -0,0 +1,153 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+var valueComparator = ottl.NewValueComparator()
```

Comment:
Please move it inside the `createIndexFunction`, just before returning the function, at line 38.

Reply:
Applied with [this commit](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/40393/commits/d695b0ea4710c5194a42dcab87c1519861e95afe)